js/adapt-stateful-session.js

```diff
@@ -1,227 +1,259 @@
 define([
   'core/js/adapt',
-  './serializers/default',
-  './serializers/questions',
-  'core/js/enums/completionStateEnum'
-], function(Adapt, serializer, questions, COMPLETION_STATE) {
-
-  // Implements Adapt session statefulness
-
-  var AdaptStatefulSession = _.extend({
-
-    _config: null,
-    _shouldStoreResponses: true,
-    _shouldRecordInteractions: true,
-
-    // Session Begin
-    initialize: function(callback) {
-      this._onWindowUnload = this.onWindowUnload.bind(this);
-
-      this.getConfig();
-
-      this.getLearnerInfo();
-
-      // Restore state asynchronously to prevent IE8 freezes
-      this.restoreSessionState(function() {
-        // still need to defer call because AdaptModel.check*Status functions are asynchronous
-        _.defer(this.setupEventListeners.bind(this));
-        callback();
-      }.bind(this));
-    },
-
-    getConfig: function() {
-      this._config = Adapt.config.has('_spoor') ? Adapt.config.get('_spoor') : false;
-
-      this._shouldStoreResponses = (this._config && this._config._tracking && this._config._tracking._shouldStoreResponses);
+  './scorm/wrapper',
+  'core/js/enums/completionStateEnum',
+  './serializers/ComponentSerializer',
+  './serializers/SCORMSuspendData'
+], function(Adapt, ScormWrapper, COMPLETION_STATE, ComponentSerializer, SCORMSuspendData) {
+
+  class StatefulSession extends Backbone.Controller {
+
+    initialize() {
+      _.bindAll(this, 'beginSession', 'onVisibilityChange', 'endSession');
+      this.scorm = ScormWrapper.getInstance();
+      this._trackingIdType = 'block';
+      this._componentSerializer = null;
+      this._shouldStoreResponses = true;
+      this._shouldRecordInteractions = true;
+      this.beginSession();
+    }
 
-      // Default should be to record interactions, so only avoid doing that if _shouldRecordInteractions is set to false
-      if (this._config && this._config._tracking && this._config._tracking._shouldRecordInteractions === false) {
+    beginSession() {
+      this.listenTo(Adapt, 'app:dataReady', this.restoreSession);
+      this._trackingIdType = Adapt.build.get('trackingIdType') || 'block';
+      this._componentSerializer = new ComponentSerializer(this._trackingIdType);
+      // suppress SCORM errors if 'nolmserrors' is found in the querystring
+      if (window.location.search.indexOf('nolmserrors') !== -1) {
+        this.scorm.suppressErrors = true;
+      }
+      const config = Adapt.spoor.config;
+      if (!config) return;
+      const tracking = config._tracking;
+      this._shouldStoreResponses = (tracking && config._tracking._shouldStoreResponses);
```

Comment:
Should be `tracking && tracking._shouldStoreResponses`.

Reply:
done in https://github.com/adaptlearning/adapt-contrib-spoor/pull/197/commits/6fe523b804c56521147c4358729dd9fcc2c6fb11